### PR TITLE
Updates ServiceBusClientBuilder to match API changes

### LINKED ISSUE #9401: [SB Track2] Update ServiceBusClientBuilder to new API design.
https://apiview.dev/Assemblies/Review/6af850ed7eb04843a6897afbe13cdd28

### PR DESCRIPTION
**Pull request by conniey**
* Updates service bus builder to match new API changes
  * Adds ServiceBusReceiverClientBuilder and ServiceBusSenderClientBuilder
* ReceiveMessageOptions is no longer a public class.
* Adds synchronous classes
* Cleans up samples.
* Removes overlapping samples.
* Fixes build issues.

API View: https://apiview.dev/Assemblies/Review/6af850ed7eb04843a6897afbe13cdd28

Fixes #9401

### CODE REVIEW

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ReceiveMessageOptions.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
```
Comment (hemanttanwar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ReceiveMessageOptions.java]: `ReceiveMessageOptions` :  Should this be removed ? You have added these properties in  ServiceBusReceiverBuilder already.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java [hunk 2]
```diff
@@ -140,90 +148,77 @@ public ServiceBusClientBuilder proxyOptions(ProxyOptions proxyOptions) {
...
+     * @return The updated {@link ServiceBusClientBuilder} object.
      */
     public ServiceBusClientBuilder transportType(AmqpTransportType transportType) {
         this.transport = transportType;
         return this;
     }
 
     /**
-     * @param retryOptions to manage AMQP connection.
+     * A new instance of {@link ServiceBusSenderClientBuilder} used to configure Service Bus message producers.
```
Comment (hemanttanwar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java]: Should we use term `producers` ? This is eventhub term. How about `senders`

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java [hunk 3]
```diff
@@ -338,4 +320,252 @@ private ProxyOptions getDefaultProxyConfiguration(Configuration configuration) {
...
+
+        /**
+         * Sets the maximum duration within which the lock will be renewed automatically. This value should be greater
+         * than the longest message lock duration.
+         *
+         * @param renewDuration The maximum duration within which the lock will be renewed automatically.
+         *
+         * @return The modified {@link ServiceBusReceiverClientBuilder} object.
+         */
+        public ServiceBusReceiverClientBuilder maxAutoLockRenewalDuration(Duration renewDuration) {
```
Comment (hemanttanwar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java]: I thought of changing this in APIview
renewDuration->renewalDuration

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ReceiveMessageOptions.java (hunk 1 above)
Comment (conniey) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ReceiveMessageOptions.java]: This is a package-private class. It's easier than passing around 6 parameters.

Comment (hemanttanwar) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ReceiveMessageOptions.java]: Okay, Its fine. This package-private. No need to take it out.

**Pull request merged**